### ISSUE TITLE
problem with tag based acl policy

### ISSUE THREAD

**Issue by czytom** [Requarks/wiki#4011]
I'm unable to setup permissions using tag based policy. I added tag public for all articles, set acl to tag matches public - after that I'm unable to see wiki navigation tree and on every page reload got red popup alert. I'm only able to access articles by path

**Comment by nuraeil** [comment]
> I'm unable to setup permissions using tag based policy. I added tag public for all articles, set acl to tag matches public - after that I'm unable to see wiki navigation tree and on every page reload got red popup alert. I'm only able to access articles by path

Yeah, has been happening for quite some time. I and a couple of otheres opened a bug report in the Discussions and have not yet had an answer to it. The issue seems to lay somewhere in the GraphQL endpoint in regards to data being fetched or rather how it is processed. It also affects the permissions for the comments.

#3787
#3548
#3598
#3590

The only work around currently is to ignore the Tag-Rule and use other means to setup the permissions, atleast that is what I am doing.

I believe that NGPixel is currently too busy with version 3... I hope this issue will not carry over to it tho.

@NGPixel could you maybe drop in to comment on this if your schedule allows it? If the issue is known and being fixed with version 3, then most of the linked issues could be closed. Thank you in advance!

**Comment by dipsy** [comment]
I found problem at 
line 260 of wiki/server/graph/resolvers/page.js 
```
return WIKI.auth.checkAccess(context.req.user, ['read:pages'], {
          path: r.path,
          locale: r.localeCode
        })
```

I think need replace to
```
return WIKI.auth.checkAccess(context.req.user, ['read:pages'], {
              path: r.path,
              locale: r.locale,
              tags: r.tags // Tags are needed since access permissions can be limited by page tags too
            })
```
@NGPixel, check it?

**Comment by dipsy** [comment]
Hello guys.  tags: r.tags was wrong because PageTree don't has tags relation. So I add it manually. Here two files from 2.5.274 release, put it to `server/graph/resolvers/`
[acl_tags.zip](https://github.com/Requarks/wiki/files/8007888/acl_tags.zip)
Now page tree automatically renders with two levels down (if Page in folder and in folder) it will be in menu automatically only by one tag rule.
Wikijs Amazing project, wait for 3 version!
